notebooks/medical-federated-learning-program/data-owners/utils.py

```diff
@@ -227,3 +230,52 @@ def submit_credentials(credentials):
     except Exception:
         pass
     print("Failed to submit Data Scientist credentials. Please copy and paste.")
+
+
+def error_tracking():
+    class VarWatcher(object):
+        def __init__(self, ip):
+            self.shell = ip
+            self.id = uuid.uuid1()
+            self.sent = False
+
+        def pre_execute(self):
+            # print("pre_execute")
+            pass
+
+        def pre_run_cell(self, info):
+            # print("pre_run_cell")
+            # print(info)
+            self.id = uuid.uuid1()
+            raw_cell = info.raw_cell
+            ip = auto_detect_domain_host_ip()
+            _ = requests.post(
```

Comment:
Should we have some error checking here otherwise if this url goes down or has some delays the users notebook experience might actually be effected.

Reply:
hmm true, I will add some checks